src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -4527,6 +4529,8 @@ Expression PrimaryExpression() #PrimaryExpression:
 
         | LOOKAHEAD( ParenthesedSelect() , {!interrupted} ) retval=ParenthesedSelect()
 
+        | LOOKAHEAD( Select() , {!interrupted} ) retval=Select()
```

Comment:
Please where is this coming from and why do we need it?
It's certainly misplaces and the Semantic Lookahead will make it terribly expensive.

Reply:
It is very possible that the performance tests fail now because of this one.

Reply:
Another example would be the following query, that won't be parsable without that change: `select * from mytable where mycol = select mycol from anothertable`